File: Programmers/hyeonyoung/위장.cpp
#include <string>
#include <vector>
#include <map>

using namespace std;

int solution(vector<vector<string>> clothes)
{
    map<string, int> kind;
    for (vector<string> cloth : clothes)
    {
        kind[cloth[1]]++;
    }

    int answer = 1;
    for (pair<string, int> k : kind)
    {
        answer *= k.second + 1;
    }

    return --answer;
}

File: Programmers/dongwan/위장.py
from itertools import combinations
def solution(clothes):
    dic = dict()
    is_dic = []
    
    for cloth,type in clothes :
        if type in is_dic :
            dic[type].append(cloth)
        else :
            dic[type] = [cloth]
            is_dic.append(type)
            
    result = 1 
    for key in dic.keys():
        result *= (len(dic[key]) +1)
        
    answer  = result -1
    
    return answer